packages/assets-controllers/src/Standards/NftStandards/ERC721/ERC721Standard.ts

```diff
@@ -91,7 +91,10 @@ export class ERC721Standard {
       address,
     );
     if (!supportsMetadata) {
-      throw new Error('Contract does not support ERC721 metadata interface.');
+      // Do not throw error here, supporting Metadata interface is optional even though majority of ERC721 nfts do support it.
+      // This change is made because of instances of NFTs that are ERC404( mixed ERC20 / ERC721 implementation).
+      // As of today, ERC404 is unofficial but some people use it, the contract does not support Metadata interface, but it has the tokenURI() fct.
+      console.error('Contract does not support ERC721 metadata interface.');
     }
     return contract.tokenURI(tokenId);
```

Comment:
In edge cases where contract is ERC721 and does not have fct `tokenURI`, we are already catching the error here https://github.com/MetaMask/core/blob/8769bd80eb9a131f9fb75ae5f85491eedfc19e62/packages/assets-controllers/src/NftController.ts#L467, the user will be able to add the nft, but the image wont be displayed.